app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php

```diff
@@ -27,37 +28,59 @@ class GetStockItemData implements GetStockItemDataInterface
      * @var StockIndexTableNameResolverInterface
      */
     private $stockIndexTableNameResolver;
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
 
     /**
      * @param ResourceConnection $resource
      * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
      */
     public function __construct(
         ResourceConnection $resource,
-        StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider
```

Comment:
Basically if MSI is support backward compatibility, this part is incompatible, as we cannot add new params in construct))